### PR TITLE
Fix logic error in Apex AMP

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Within the accelerator refactor there is a small error in the Apex logic, which results in us not calling `optimizer.step` in automatic optimization. This logic should be reversed. Regarding testing, I assume we have no convergence tests in place hence Apex AMP tests did not catch this?

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/plugins/precision/apex_amp.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def pre_optimizer_step(
         # apex amp does not support closures.
         lambda_closure()
 
-        if not pl_module.automatic_optimization:
+        if pl_module.automatic_optimization:
```
Comment (tchaton) [inline pytorch_lightning/plugins/precision/apex_amp.py:160]:
```suggestion
    def pre_optimizer_step(
        self, pl_module: LightningModule, optimizer: Optimizer, optimizer_idx: int, lambda_closure: Callable, **kwargs
    ) -> bool:
        """
        always called before the optimizer step.
        """
        # apex amp does not support closures.
        lambda_closure()

        if not pl_module.automatic_optimization:
            pl_module.trainer.call_hook("on_after_backward")
        
        optimizer.step()

        return False
```

Comment (tchaton) [reply pytorch_lightning/plugins/precision/apex_amp.py:160]: The pl_module.trainer.call_hook("on_after_backward") hook is already being called elsewhere in automatic_optimization.

**Review by tchaton — COMMENTED** [review]
It should call optimizer.step for both manual and automatic.

**Review by tchaton — CHANGES_REQUESTED** [review]
Need to be called for both

**Review by tchaton — APPROVED** [review]
LGTM !
